### ISSUE TITLE
[Admin][Orders] Creating a new payment makes the initial one "Invalid" and impossible to edit

### ISSUE THREAD

**Issue by audez** [openfoodfoundation/openfoodnetwork#9841]
## Description
In Orders details, in the Payments tab, if you create a new payment, it makes the initial one "Payment status" switch to "Invalid" and you can't edit it anymore.

## Expected Behavior
You should be able to edit the first payment even after creating a new payment. 
This is particularly bad as there is #9840 , where you have to add a payment to adjust to the full price, and this invalidates the first payment..

## Actual Behaviour
In an order detail, there's a payment in the Payments tab table. 
If you click on "Create a new payment", and save a new payment, you'll see it appears in the table. But you also see that the initial payment "Payment status" is now "Invalid", the Capture" icon disappeared, and if you click on the "Void" icon, you have an error message "Could not update the payment". 
Meaning that you can no longer act on the first payment.

And this invalid payment will appear in the client invoice.

## Steps to Reproduce
1. Pass an order on a shop
2. As the shop manager, go in the order details, Payment tab (admin/orders/R635250042/payments)
3. Click "Create a new payment" and save (it's equal if you choose the same or a different payment method)
4. See the effect on the first payment

## Before creating a new payment
![Screen Shot 2022-10-21 at 20 12 07](https://user-images.githubusercontent.com/28386648/197264745-d75d5bab-4bba-4fd9-90bf-44668892af3c.png)

## After creating a new payment: Capture icon has disappeared
![Screen Shot 2022-10-21 at 20 12 17](https://user-images.githubusercontent.com/28386648/197264816-3421da48-d5f0-4168-ac2e-6755c802aea8.png)

## You can't cancel the payment
![Screen Shot 2022-10-21 at 20 12 22](https://user-images.githubusercontent.com/28386648/197264887-d4ab43c3-1adb-4d64-bf9a-be9c60604944.png)

## What appears on the invoice
![Screen Shot 2022-10-21 at 20 22 06](https://user-images.githubusercontent.com/28386648/197265337-701c6f14-96d6-4f30-bc00-30eeeaed700d.png)

## Another order: if you first void the payment, and then create a payment, it's fine
![Screen Shot 2022-10-21 at 20 13 47](https://user-images.githubusercontent.com/28386648/197264984-aa4a4b33-483c-4bb4-b507-ad7fd2097023.png)

## Workaround
There's no workaround if you don't want to capture or void the first payment (for instance, you have a 6€ first payment that has not been done yet by the client, and you have to add a 3€ payment to reach the 9€ order amount), as in #9840 .

If you can capture the payment, then you first edit the payment, and then create a new payment. 

## Severity
bug-s3: a feature is broken but there is a workaround

**Comment by RachL** [comment]
@audez FYI I've changed: 

> You should be able to edit the first payment even after creating a new payment.

We shouldn't be able to edit a payment. That's illegal in case of an order. But I agree it's weird to consider the first payment invalid. It does not happen if you capture first, which is the workaround we need to tell users.

IMO the fix should be to warn the user they must capture first. Not capture by default.

**Comment by audez** [comment]
Yes by edit I mean: capture or void the payment, same way you do in reverse

**Comment by audez** [comment]
I dont understand but i think i dont understand/know how the payment currently works in case of adjustments

But if it's mandatory to capture/void the initial payment before creating a new payment i agree that we should warn the user and then not allow them to create a new payment (deactivate the button?)

**Comment by RachL** [comment]
> not allow them to create a new payment

Let's keep it, it's a shortcut in case you don't want to capture the first once, but cancel it (current default case).

**Comment by audez** [comment]
OK so then if we do that we should warn them on click on "Create a new payment" that this will invalidate the initial payment, and don't display the "Void" icon on the initial invalidated payment

Also I think there is a real problem: it's not only for the first new payment that you create, it happens each time you will create a new payment. Creating a new payment will always invalidate the previous uncaptured one.
So for instance: I have a buyer that did a 50€ order. She paid me with 2 checks of 10 and 40, and I want to save it in the payment. I can't create 2 pending payments  of 10 and 40 :/

**Comment by RachL** [comment]
> Creating a new payment will always invalidate the previous uncaptured one.

yes, but this is logical.


> I can't create 2 pending payments of 10 and 40 :/

Yes, theses cases exist in case of subscription. But do not fit the model of an ecommerce order. Currently our order model is not made for these cases, it's not a bug, it's something we need new feature for.

**Comment by audez** [comment]
> yes, but this is logical.

But then I don't understand this:
- There's an initial order of 10€
- The producer makes an adjustment and the order becomes 12€
- if creating a new payment always invalidate the previous one, why we don't automatically invalidate the 10€ payment intent, and create a new of 12€? Instead of asking the producer to do it manually.. 

I think it's very confusing for producers, because first we ask them to create the payments themselves, then to guess that they have to capture or void the initial one and only AFTER that create a new one. Although they might not have received any payment yet. 
How to explain that they can't create 2 payments and capture both after..which is actually the same...

Also it raises the issue discussed [here](https://openfoodnetwork.slack.com/archives/CG7NJ966B/p1648041940781449?thread_ts=1647503746.206969&cid=CG7NJ966B) by @tschumilas: from the Orders list page, the capture button is displayed next to the new price, so the producer will click on "Capture" thinking that they're capturing the amount owed, but actually they just captured the initial amount..

For instance:
- A buyer makes an initial 4€ order 
![Screen Shot 2022-10-24 at 14 00 58](https://user-images.githubusercontent.com/28386648/197521641-52413b98-1924-44e0-ae49-9c4abfd8c5b5.png)

- The producer changes the weight of a product in the BOM page, and the new order total is now 12€. 
- On the Orders list page, the producer clicks on "Capture" button, thinking them capture the amount owed
![Screen Shot 2022-10-24 at 14 01 36](https://user-images.githubusercontent.com/28386648/197521703-bb4b926f-ede5-48bd-ace2-5983f0b4723e.png)

- But it actually only captured the initial payment..
![Screen Shot 2022-10-24 at 14 01 49](https://user-images.githubusercontent.com/28386648/197521808-e38d94d9-444c-41eb-9ba0-118ac9581465.png)

**Comment by RachL** [comment]
> if creating a new payment always invalidate the previous one, why we don't automatically invalidate the 10€ payment intent, and create a new of 12€? Instead of asking the producer to do it manually..

Because there are several use cases at play. Perhaps the edition or the order was made _after_ the first real life payment. And the producer just forgot to capture it at that time. We don't know that. 

> But it actually only captured the initial payment..

Again logical from a system point of vue. The system can only capture a payment that exist.

All we can do here is add messages, we cannot force a default behavior.

**Comment by tschumilas** [comment]
I think we need to have the concept of a payment intent.   Right now -  everything is just a 'payment' -   whether it is automated or cash.   But when a consumer selects a cash (non-automated) payment method at checkout,   they are making a payment intent -  not a payment.   A cash method payment intent is a payment once the cash is received.   And I concur it looks very bad that 'invalid' always shows beside cash payment methods on invoices - I get customers asking me all the time what was wrong with their payment.  The only option right now is for the hub/seller  to go through all the orders and void all the cash payments before they begin the order management process.  Its very time consuming.

**Comment by audez** [comment]
thanks for the input @tschumilas and I think it rings with the discussion in the other issue https://github.com/openfoodfoundation/wishlist/issues/311#issuecomment-1288794509

**Comment by RachL** [comment]
This need further work before it can be picked-up by a dev: moving to wishlist
